pkg/crds/remove.go

```diff
@@ -0,0 +1,385 @@
+package crds
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	cephv1 "github.com/rook/rook/pkg/client/clientset/versioned/typed/ceph.rook.io/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/retry"
+)
+
+var confirmationDeleteCRDPatch = map[string]interface{}{
+	"spec": map[string]interface{}{
+		"cleanupPolicy": map[string]string{
+			"confirmation": "yes-really-destroy-data",
+		},
+	},
+}
+
+type RemoveCustomResourceDefinitions struct {
+	rookCeph         cephv1.CephV1Interface
+	clusterNamespace string
+	jsonPatch        []byte
+}
+
+func New(clientsets *k8sutil.Clientsets, clusterNamespace string) *RemoveCustomResourceDefinitions {
+	jsonPatch, _ := json.Marshal(confirmationDeleteCRDPatch)
+	clientsets.Rook.CephV1()
+
+	return &RemoveCustomResourceDefinitions{
+		rookCeph:         clientsets.Rook.CephV1(),
+		clusterNamespace: clusterNamespace,
+		jsonPatch:        jsonPatch,
+	}
+}
+
+func (r *RemoveCustomResourceDefinitions) patch(ctx context.Context, resourceName string) error {
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		_, err := r.rookCeph.CephClusters(r.clusterNamespace).Patch(ctx,
+			resourceName, types.MergePatchType, r.jsonPatch, v1.PatchOptions{})
+		return err
+	})
+}
+
+func (r *RemoveCustomResourceDefinitions) removeCephClusters(ctx context.Context) error {
+	cephClusters, err := r.rookCeph.CephClusters(r.clusterNamespace).List(ctx, v1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, resource := range cephClusters.Items {
+		err = r.patch(ctx, resource.Name)
+		if err != nil {
+			return err
+		}
+
+		err := r.rookCeph.CephClusters(r.clusterNamespace).Delete(ctx, resource.Name, v1.DeleteOptions{})
```

Comment:
If the CRs are only deleted, they will likely stay in pending because of the finalizer. So after you delete the CR, check if it still exists, and if so, update it to remove the finalizer. Or if there is any issue with that approach, you could remove the finalizer before deleting it.